### ISSUE TITLE
Error with binary files

### ISSUE THREAD

**Issue by dfishburn** [jreybert/vimagit#60]
If I try to add a .zip file using S in the Magit window I get the following:

```
Error detected while processing function magit#stage_hunk[32]..magit#stage_block[11]..magit#git#git_apply:
line    8:
Git error: error: cannot apply binary patch to 'sqlanywhere/test/testdata_064A.zip' without full index line^@error: sqlanywhere/test/testdata_064A.zip: patch does not apply
line    9:
Git cmd: "C:\Program Files\Git\cmd\git.EXE" apply --recount --no-index --cached -
line   10:
Tried to aply this
line   11:
['diff --git a/sqlanywhere/test/testdata_064A.zip b/sqlanywhere/test/testdata_064A.zip', 'index dfb1dd7..6184e0a 100644', 'Binary files a/sqlanywhere/test/testdata_064A.zip and b/sqlanywh
ere/test/testdata_064A.zip differ', '']
```

**Comment by jreybert** [comment]
Hi dfishburn,

could you give me the output of the command:

```
:!file --mime sqlanywhere/test/testdata_064A.zip
```

Thanks

**Comment by dfishburn** [comment]
The file utility does not exist on Windows.
I couldn't find any good alternatives for it either.

But, this is just a standard Zip file I created with the 7-zip open source utility.

**Comment by jreybert** [comment]
Well, that's a shame I do not check the result of system("file...") command...

I'll try to figure a way to detect binary files on windows. No matter that is just a simple zip file, if vimagit does not detect that a modified file is a binary, it will fail to stage this modified file.

**Comment by jreybert** [comment]
May also fix #61

**Comment by jreybert** [comment]
Hi @dfishburn , could you please try the branch dev/git_diff and tell me if it fixes this issue?
